build-release-tools/application/jump_version.py

```diff
@@ -96,24 +100,61 @@ def update_changelog(self, message=None):
         # -b: Force a version to be less than the current one
         # -m: Don't change (maintain) the trailer line in the changelog entry; i.e.
         #     maintain the maintainer and date/time details
-        cmd_args = ["dch", "-v", self._version, "-b", "-m"]
-        if message is None:
-            message = "new release {0}".format(self._version)
-        cmd_args += ["-p", message]
-        proc = subprocess.Popen(cmd_args,
-                                cwd=self._repo_dir,
-                                stderr=subprocess.PIPE,
-                                stdout=subprocess.PIPE,
-                                shell=False)
-        (out, err) = proc.communicate()
-
-        if linked:
-            os.remove(os.path.join(self._repo_dir, "debian"))
-
-        if proc.returncode != 0:
+        try:
+            cmd_args = ["dch", "-v", self._version, "-b", "-m"]
+            if message is None:
+                message = "new release {0}".format(self._version)
+            cmd_args += ["-p", message]
+            common.run_command(cmd_args, directory=self._repo_dir)
+      
+            if linked:
+                os.remove(os.path.join(self._repo_dir, "debian"))
+
+            return True
+        except Exception, err:
             raise RuntimeError("Failed to add an entry for {0} in debian/changelog due to {1}".format(self._version, err))
 
-        return True
+
+class NPMVersionUpdater(object):
```

Comment:
Forget the comments above.

Because I saw several(two) ***Updater classes, so I think if we can improve their encapsulation.

Class like Gitbits, NPM..etc. just run their cmds. 
And ***Updater do operation on repos' files.
So if all Updater or other class do actions on repos could be packaged into one class ```RepoFileOperator```?